### PR TITLE
Update Heatmap legend after style changes

### PR DESCRIPTION
**Pull request by VictorVelarde**
Related to: https://github.com/CartoDB/cartodb/issues/13763 and https://github.com/CartoDB/onpremises/issues/518
---

- [x] Dev
- [x] Frontend CR
- [x] Deployment to staging
- [x] Acceptance
- [ ] Deployment to production

# Acceptance
- Create a `New map` from a dataset with points
- Go to the layer and then go to the `Style` view.
- Change the Aggregation to `Heatmap`
- Go to the `Legend` view.
- Add a `Custom Chloropleth Legend`
- Go back to the `Style` view and change the color ramp.
- Colorbar at `Legend` must be updated with new color properties

### CODE REVIEW

File: lib/assets/javascripts/builder/helpers/style.js [hunk 1]
```diff
@@ -37,6 +37,19 @@ module.exports = {
     return size;
   },
 
+  getColorsFromRange: function (styleModel) {
```
Comment (ivanmalagon) [inline lib/assets/javascripts/builder/helpers/style.js]: A spec for this is missing.

File: lib/assets/javascripts/builder/helpers/style.js [hunk 2]
```diff
@@ -37,6 +37,19 @@ module.exports = {
...
+    var color = this.getColor(styleModel);
+
+    if (!(color && color.range)) { return []; }
+    var range = color.range.length;
+
+    return _.range(range).map(function (v, index) {
```
Comment (ivanmalagon) [inline lib/assets/javascripts/builder/helpers/style.js]: If `color.range` is already an array, why creating the `range`?

File: lib/assets/javascripts/builder/helpers/style.js (hunk 1 above)
Comment (VictorVelarde) [reply lib/assets/javascripts/builder/helpers/style.js]: ok, in fact it seems there is no spec file for this helpers

File: lib/assets/javascripts/builder/helpers/style.js (hunk 2 above)
Comment (VictorVelarde) [reply lib/assets/javascripts/builder/helpers/style.js]: Very true, I took another helper as an example. I have refactored both of them

**Comment by ivanmalagon** [comment]
Acceptance passed!

**Comment by VictorVelarde** [comment]
Shipped and test at production

**Pull request merged**